CDCgov/prime-reportstream
# 1416 Add folders to organize files in Blob Store

This PR adds two levels of folders for received, internal and send files as follows:
received/<org name>.<sender name>/file[.hl7|.csv]
internal/<org name>.<receiver name>/file.internal.csv
ready/<org name>.<receiver name>/file[.hl7|.csv]

Tested the UI changes by submitting data on both this branch and master's router to generate blobs with and without the folders, then ran the UI to download both files.

Test:
1. Use the `./gradlew testend2end` to send data and verify the test passes
2. Look in the blob store to verify the folders are generated as expected
3. Run the `./gradlew primeCLI --args="test --run compare"` and verify the test passes
4. Send the simplereport test data and login to http://localhost:7071/api/download and verify you can download the data files as expected.  E.g. `curl -X POST -H "client:simple_report"  -H "content-type:text/csv" --data-binary "@./src/test/csv_test_files/input/simplereport.csv"  "http://localhost:7071/api/reports"`

## Changes
- Added folders to the blob store
- Updated compare test files to pass due to changes in master
- Updated front end to remove blob folders from filename when present

## Checklist

### Testing

### Process

## Fixes
- #1010 
- #1416
